### DISCUSSION TITLE
Show test coverage in the gutter

### DISCUSSION THREAD

**Discussion by eth0net** [zed-industries/zed#24002] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

One thing I miss from other editors is the ability to show test coverage indicators in the gutter, in a similar way to git status indicators.

To differentiate the test coverage from git status they would need to be a different colour and/or displayed somewhere slightly differently, perhaps to the right of line numbers instead. For reference, JetBrains shows it on the right of the line numbers as well.

I would be interested in working on this as I would like to make more contributions, but I wanted to open this issue to track it and perhaps get some tips on where to look for the relevant git status indicator code.

#### Git status in Zed
![zed git](https://github.com/user-attachments/assets/3b010eae-56b1-4f75-a81c-4f5f6a7cd5c0)

#### Git status in JetBrains
![jetbrains git](https://github.com/user-attachments/assets/ab35e9f4-b3aa-4880-9263-b7003985e7b7)

#### Coverage in JetBrains
![jetbrains coverage](https://github.com/user-attachments/assets/0537b13a-5c7b-403c-80d7-f3fc5549c8ba)

**Comment by hhamud** [comment]
any updates on this?

**Comment by AndroxxTraxxon** [comment]
I'm pretty happy with the way that VS Code displays this -- with Git changes on one side of the line numbers, and coverage indicators on the other. In this screenshot, the git change indicators are on the right side of the line number, while the coverage indicators are on the left. 
![Screenshot 2025-04-30 at 3 05 02 PM](https://github.com/user-attachments/assets/64f88308-1020-4c3e-a503-652d02357da8)

**Comment by tv42** [comment]
Here's the relevant VSCode extension, for comparison: https://marketplace.visualstudio.com/items?itemName=ryanluker.vscode-coverage-gutters

**Comment by w-charlton-virifides** [comment]
found myself switching off to cursor/vscode for test development for this reason

**Comment by jeremybanka** [comment]
Now that the debugger is working well with Vitest, coverage visualization is the number one thing that brings me back to VSCode also

**Comment by danhawkins** [comment]
Same here, I would be useing zed full time if only it had this feature

**Reply by ghost** [reply]
Same here!

**Comment by FredrikAugust** [comment]
Second this! It would be an amazing feature. Is this on the roadmap, or has anyone begun implementation of this?

**Comment by ArtemKart** [comment]
Any updates?

**Comment by 0pilatos0** [comment]
+1 One this feature! would love to see something like this

**Comment by nyzss** [comment]
would love to see this!
